src/session_concurrency/session_concurrency_manager.cc

```diff
@@ -141,31 +142,36 @@ SessionConcurrencyManager::SendSequentialRequestsForOneSession(
   rapidjson::Document chat_history(rapidjson::kArrayType);
 
   for (size_t i{0}; i < one_session_payloads.size(); ++i) {
-    const size_t payload_index{one_session_payloads[i]};
+    const size_t payload_dataset_index{one_session_payloads[i]};
 
-    request_records.emplace_back();
-
-    auto& request_record{request_records.back()};
+    auto& request_record{request_records.emplace_back()};
 
     request_handler_->SendRequestAndWaitForResponse(
-        payload_index, chat_history, request_record);
+        payload_dataset_index, chat_history, request_record);
 
     const bool is_last_request{i == one_session_payloads.size() - 1};
 
     if (is_last_request) {
       break;
     }
 
-    GetAndWaitForDelayMs(payload_index);
+    GetAndWaitForDelay(payload_dataset_index);
   }
 }
 
 void
-SessionConcurrencyManager::GetAndWaitForDelayMs(size_t payload_index) const
+SessionConcurrencyManager::GetAndWaitForDelay(
+    size_t payload_dataset_index) const
 {
-  const auto delay{payload_dataset_manager_->GetDelayForPayload(payload_index)};
+  const auto delay{payload_dataset_manager_->GetDelay(payload_dataset_index)};
+
+  if (!delay) {
+    throw std::runtime_error(
+        "Missing 'delay' input for payload with index " +
+        std::to_string(payload_dataset_index));
+  }
```

Comment:
As mentioned, since GetDelay() in PayloadDatasetManager can now return optional for missing delay, this is the new place where an expected delay that's missing causes an exception to be thrown.